Add RELEASE_LOG_FAULT when decoding child class of allowed class through _WKRemoteObjectRegistry

achristensen07 opened the pull request:
#### f50839f0e2654841a96d10e974fa1a326641cbbc
<pre>
Add RELEASE_LOG_FAULT when decoding child class of allowed class through _WKRemoteObjectRegistry
<a href="https://bugs.webkit.org/show_bug.cgi?id=254776">https://bugs.webkit.org/show_bug.cgi?id=254776</a>
rdar://106810634

Reviewed by David Kilzer.

* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
(alwaysAllowedClasses):
(checkIfClassIsAllowed):
* Source/WebKit/Shared/API/Cocoa/_WKRemoteObjectInterface.mm:
(initializeMethod):
(isContainerClass): Deleted.
(propertyListClasses): Deleted.

Canonical link: <a href="https://commits.webkit.org/262407@main">https://commits.webkit.org/262407@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/f623aadaaa73fb966bedf121d941a4f1debd3e4e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [❌ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1367 "1 style error") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1400 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/1447 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2266 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/1216 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/1457 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/1460 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1359 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1379 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1273 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1250 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2113 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1284 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1235 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1213 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1246 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1270 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2342 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1313 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/1174 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1227 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1242 "Passed tests") | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1335 "Built successfully") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/158 "Built successfully and passed tests") | | | | 
<!--EWS-Status-Bubble-End-->

ddkilzer commented on Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
```diff
@@ -783,6 +801,11 @@ static void checkIfClassIsAllowed(WKRemoteObjectDecoder *decoder, Class objectCl
+    if (alwaysAllowedClasses().contains((__bridge CFTypeRef)objectClass))
+        return;
+
+    RELEASE_LOG_FAULT(IPC, "Unexpected class %s", NSStringFromClass(objectClass).UTF8String);
```
If you always want this logged, create a separate channel other than `IPC`.  (I would recommend doing this.)

ddkilzer reviewed (approved):
r=me, but consider creating a separate channel for this instead of reusing `IPC` so this issue is always logged.

achristensen07 committed:
Add RELEASE_LOG_FAULT when decoding child class of allowed class through _WKRemoteObjectRegistry
https://bugs.webkit.org/show_bug.cgi?id=254776
rdar://106810634

Reviewed by David Kilzer.

* Source/WebKit/Shared/API/Cocoa/WKRemoteObjectCoder.mm:
(alwaysAllowedClasses):
(checkIfClassIsAllowed):
* Source/WebKit/Shared/API/Cocoa/_WKRemoteObjectInterface.mm:
(initializeMethod):
(isContainerClass): Deleted.
(propertyListClasses): Deleted.

Canonical link: https://commits.webkit.org/262407@main
